imsim/skycat.py

```diff
@@ -171,7 +155,7 @@ def getObj(self, index, gsparams=None, rng=None, exp_time=30):
         -------
         galsim.GSObject
         """
-        if not self.objects:
+        if not self._objects:
```

Comment:
I think this should be `objects`, not `_objects`.  In practice, it might not matter, since probably `getNObjects` will always be called before `getObj`.  But I think logically, this test should be if the loaded object list is empty, not if the objects haven't been loaded from the skyCatalog yet.

Reply:
I guess it is clearer to test `objects`, but my thought process was that `_objects` would be an empty list if skyCatalogs didn't find any objects within the search region.  Anyway, I'll make the change.

Reply:
But it could also still be None if you haven't checked yet.